oqpy/program.py

```diff
@@ -195,12 +195,14 @@ def _add_subroutine(self, name: str, stmt: ast.SubroutineDefinition) -> None:
         """
         self.subroutines[name] = stmt
 
-    def _add_defcal(self, qubit_name: str, name: str, stmt: ast.CalibrationDefinition) -> None:
+    def _add_defcal(
+        self, qubit_names: list[str], name: str, stmt: ast.CalibrationDefinition
+    ) -> None:
         """Register a defcal which has been used.
 
         Defcals are added to the top of the program upon conversion to ast.
         """
-        self.defcals[(qubit_name, name)] = stmt
+        self.defcals[(*qubit_names, name)] = stmt
```

Comment:
Preferable to keep these separate I think: I would just make it of form `tuple[list[str], str]`

```suggestion
        self.defcals[(qubit_names, name)] = stmt
```

Reply:
list is not hashable so I transformed in tuple. One drawback is the comma with single-element tuple.